### PR TITLE
Cookie validation

### PR DESCRIPTION
**Pull request by AnuguRakeshReddy**

### CODE REVIEW

File: Oauth/routes/index.js [hunk 1]
```diff
@@ -120,51 +120,276 @@ router.get('*', (req, res) => {
...
   }
+  else {
+    return clientid;
+  }
+
 }
 
+
+const redis = require("redis");
+const jwt = require("jsonwebtoken");
```
Comment (Abhijithchintu) [inline Oauth/routes/index.js:155]: add this at the top of the file

File: Oauth/routes/index.js [hunk 2]
```diff
@@ -120,51 +120,276 @@ router.get('*', (req, res) => {
...
+
+
+  var siftvalue = await rediscl.get(uid);
+  console.log(siftvalue, " here it is");
   return res.send(" You have successfully logged in");
+
 });
+
+// Let's define a helper function that we will use in most of our routes.
+function validate_jwt(req, res) {
```
Comment (Abhijithchintu) [inline Oauth/routes/index.js:224]: add comments above function

File: Oauth/routes/index.js [hunk 3]
```diff
@@ -120,51 +120,276 @@ router.get('*', (req, res) => {
...
+  var charset = "abcdefghijklmnopqrstuvwxyz0123456789";
+
+  for (var i = 0; i < len; i++)
+    text += charset.charAt(Math.floor(Math.random() * charset.length));
+
+  return text;
+}
+
+
+router.post("/profile", (req, res, next) => {
```
Comment (Abhijithchintu) [inline Oauth/routes/index.js:318]: add comments clearly and above function

**Review by Abhijithchintu — CHANGES_REQUESTED** [review]
complete this asap

File: Oauth/routes/index.js [hunk 4]
```diff
@@ -120,51 +120,276 @@ router.get('*', (req, res) => {
...
+    user_id = await validateLogin(req);
+    console.log(user_id) + "Here is it!!!";
+
   } catch (error) {
     logger.error("This is Login error", error);
     return res.send(error.message);
   }
   logger.info("User details are correct and successfully logged in");
+
+  // Generate new refresh token and it's expiration
```
Comment (manu156) [inline Oauth/routes/index.js]: remove comments

File: Oauth/routes/index.js [hunk 5]
```diff
@@ -120,51 +120,276 @@ router.get('*', (req, res) => {
...
+    expiresIn: jwt_expiration
+  });
+
+  // Set browser httpOnly cookies
+  res.cookie("access_token", token, {
+    // secure: true,
+    httpOnly: true
+  });
+  res.cookie("refresh_token", refresh_token, {
+    // secure: true,
```
Comment (manu156) [inline Oauth/routes/index.js]: remove unused parameters

File: Oauth/routes/index.js [hunk 6]
```diff
@@ -120,51 +120,276 @@ router.get('*', (req, res) => {
...
+
+    // Check if tokens found in cookies
+    if (accesstoken && refreshtoken) {
+
+      // They are, so let's verify the access token  
+      jwt.verify(accesstoken, jwt_secret, async function (err, decoded) {
+
+        if (err) {
+
+          // There are three types of errors, but we actually only care
```
Comment (manu156) [inline Oauth/routes/index.js]: remove all these comments. comments should only be for functions

**Comment by AnuguRakeshReddy** [comment]
> complete this asap

resolved

**Review by Abhijithchintu — APPROVED** [review]

**Pull request merged**